[AArch64] Change the default vscale-for-tuning to 1.

davemgreen opened the pull request:
Most AArch64 cpus outside of Neoverse V1 (256) and A64FX (512) have an SVE vector length of 128, and in environments like Android (where no mcpu option is common) we would expect all cpus to match. This patch changes the default vector length to 128 with -mcpu=generic, to match the most common case.

david-arm commented on llvm/test/Transforms/LoopVectorize/AArch64/low_trip_count_predicates.ll:
Could you add `-mcpu=neoverse-v1` to the RUN line here? When I wrote these tests I was wrongly assuming a default of vscale=2 and so a test like `low_vf_ic_is_better` is trying to show how the vectoriser deliberately doesn't choose the max possible.

david-arm reviewed (approved):
LGTM!

davemgreen committed:
[AArch64] Change the default vscale-for-tuning to 1.

Most AArch64 cpus outside of Neoverse V1 (256) and A64FX (512) have an SVE
vector length of 128, and in environments like Android (where no cpu option is
common) we would expect all cpus to match. This patch changes the default
vector length to 128 with -mcpu=generic, to match the most common case.
